### DISCUSSION TITLE
GLMakie and Julia on CoCalc

### DISCUSSION THREAD

**Discussion by mkrishan** [sagemathinc/cocalc#6310] [Q&A]
CoCalc's different ecosystems for Julia such as Jupyter/VSCode/linux kernels can't update GLMakie latest version. I can only find v0.4.0 or v0.4.4: which is so outdated. Maybe I am missing something?

Question: How to (force) update GLMakie in web-based CoCalc in all of the above?

Thank you for your time.

**Comment by haraldschilly** [comment]
There is certainly a way to install julia packages independently of the globally installed ones. By default, the projects point to a global depot. Check this environment variable:

```
$ echo $JULIA_DEPOT_PATH 
/home/user/.julia:/ext/julia/depot/
```

I never really understood how julia's packaging system works. This is a way how to provide users with at least some installed packages, even when their project's don't have internet access. 

You could set that DEPOT PATH somewhere else in your project or work with other ways how julia does this (I don't know much about this, as mentioned above).

If you also want to use a jupyter kernel, you can create your own in your project as well. Just copy the global kernel from `/ext/jupyter/kernels/julia-1.8/kernel.json` to get started and follow these notes analogous: https://doc.cocalc.com/howto/custom-jupyter-kernel.html

Regarding GLMakie itself, I just tried to update it on our staging server behind the scenes, but it's still just `v0.4.4`. Do you have any idea what's going on? [Their github page here mentions to add it as usual](https://github.com/MakieOrg/Makie.jl#installation). My guess is there are other dependencies that lock it to an older version. I don't know how to untangle that. In any case, installing it in its own isolated package depot should help.

**Reply by williamstein** [reply]
> I never really understood how julia's packaging system works. 

Whenever I've tried to google for help on julia packaging, I've felt confusing and contradictory documentation.  I think this is partly due to them rewriting it to make it different several times...
